### PR TITLE
CIP-0121? | Integer-ByteString conversions

### PR DESCRIPTION
**Pull request by kozross**
CIP for Plutus primops for converting `BuiltinInteger` to `BuiltinByteString`, and back.

[Human-readable version](https://github.com/mlabs-haskell/CIPs/tree/koz/to-from-bytestring/CIP-0121).

### CODE REVIEW

File: CIP-0087/CIP-0087.md
Comment (rphair) [inline CIP-0087/CIP-0087.md:2]:
```suggestion
CIP: ?
```
Choosing a CIP number in advance of a review is always a problem (collisions with CIPs in progress & sometimes online discussions are almost assured) as well as a specific problem in this case:
- https://github.com/cardano-foundation/CIPs/pull/440

Please @kozross also make a link to the readable version of your proposal in your OP (https://github.com/cardano-foundation/CIPs/pull/624#issue-2005257354) after you've
- renamed your containing directory to something without a number (e.g. `CIP-XXXX` or `CIP-Integer-ByteString`). 
- like all current CIPs, renamed the file itself to `README.md`

**Review by rphair — COMMENTED** [review]
Thanks @kozross for a well documented submission.  First I'm retitling this by convention the same as the current document title (_Integer-ByteString conversions_) especially since the more succinct title will work better in our CIP list once merged.

The main item I believe needs to be fixed (or checked at least) are the CIP-0035 requirements for builtin changes.  Though different language is used, I think the general requirements are satisfied: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0035/README.md#processes

... but not all the particular requirements: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0035/README.md#additions-to-the-plutus-core-builtins

@michaelpj any time you can look over this submission I would really appreciate it.  If I'm wrong about this not meeting the particular specifications for builtins by CIP-0035 I would happily stand corrected.  To me this looks ready for acceptance as soon as all the formalities are met.

**Review by michaelpj — APPROVED** [review]

**Comment by zliu41** [comment]
@rphair Can we assign this a CIP number and merge it? Not only has it already been approved, but it has already been implemented and released in Plutus V3.

**Comment by rphair** [comment]
@zliu41 my apologies for being under the impression that the currently debated https://github.com/cardano-foundation/CIPs/issues/794 was relative to CIP-0058 alone; in fact this one had fallen through the cracks in the belief it was still waiting for input from the Plutus team.

This PR branch is on an MLabs repository so I'll propose some changes there which @kozross can approve, to  add a CIP number `121` and update the CIP repository `README`, so we don't have to wait for the next CIP meeting.  Once that's committed then I'll approve this PR & expedite confirmations from other editors: shouldn't take too long after that.

File: CIP-XXXX/README.md
Comment (rphair) [inline CIP-XXXX/README.md:2]:
```suggestion
CIP: 121
```

File: CIP-XXXX/README.md
Comment (rphair) [inline CIP-XXXX/README.md:5]:
```suggestion
Status: Active
```
As per https://github.com/cardano-foundation/CIPs/pull/624#issuecomment-2091414191.

File: CIP-XXXX/README.md
Comment (rphair) [inline CIP-XXXX/README.md:528]:
```suggestion
### Acceptance Criteria

- [x] Implemented and released in Plutus V3.

We consider the following criteria to be essential for acceptance:
```
As per https://github.com/cardano-foundation/CIPs/pull/624#issuecomment-2091414191.

**Review by rphair — COMMENTED** [review]
@kozross I'll also submit a PR against this branch to modify the `README` for the CIP repository that officially assigns the CIP number (otherwise we wait for the 2-week meeting intervals).

File: CIP-XXXX/README.md
Comment (Ryun1) [reply CIP-XXXX/README.md:528]: Are we counting this as "released" when it is on mainnet or on a testnet?

Comment (rphair) [reply CIP-XXXX/README.md:528]: Although I was thinking in either case the `Active` status would apply, maybe the wording here should be _included_ (in Plutus V3) if the term "released" here isn't according to common usage.  cc @zliu41 @michaelpj

Comment (zliu41) [reply CIP-XXXX/README.md:528]: > I was thinking in either case the Active status would apply

Yeah, I think that's right. It has been released in `plutus-core` a number of versions ago, and it has also been included in the latest node pre-release for SanchoNet. It won't be on mainnet until after the Conway HF.

Comment (rphair) [reply CIP-XXXX/README.md:528]: @Ryun1 it does sound like the above qualifies it as a "release", since people will already be building on it; let us know if that's OK with you.

Comment (Ryun1) [reply CIP-XXXX/README.md:528]: Sounds good !

Comment (rphair) [reply CIP-XXXX/README.md]: @kozross (cc @zliu41) if you can admit this language, or equivalent, into the CIP I would be ready to approve it.  I think the addition is important because it demonstrates to the general community (not just Plutus architects) why this CIP is being merged with `Active` status.

Comment (kozross) [reply CIP-XXXX/README.md]: I added text to this effect to the section immediately following.

Comment (rphair) [reply CIP-XXXX/README.md]: thanks; I see it now.

File: README.md
Comment (rphair) [inline README.md:142]:
```suggestion
```

**Review by rphair — COMMENTED** [review]
@kozross look ready to merge from an editorial perspective, except for one artefact in the repo `README` probably from rebasing or fixing a merge conflict; will approve this when that's fixed...

**Review by rphair — APPROVED** [review]

**Comment by rphair** [comment]
@Ryun1 @Crypto2099 I'm marking this as `Last Check` since it seems that the same review process as https://github.com/cardano-foundation/CIPs/pull/806 has been followed as noted in https://github.com/cardano-foundation/CIPs/pull/806#pullrequestreview-2066242152 (cc @kozross) and so putting it on the agenda as such (https://hackmd.io/@cip-editors/89).

Likewise @michaelpj @zliu41 please let us know if any technical reservations.  It seems to me everything above has been addressed & I would be in favour of merging these 2 related proposals before the meeting if the other editors agree.

File: README.md
Comment (rphair) [reply README.md:150]: @Ryun1 this was fixed for later candidate CIP-0122 in https://github.com/cardano-foundation/CIPs/pull/806#discussion_r1606791305 but I missed it here.  I think @kozross was following an apparent precedent of https://github.com/cardano-foundation/CIPs/pull/624#issuecomment-2092353644 which modified the `README` because CIP-0058 had taken so long to merge & was referred to by other actively reviewed documents.  TL;DR yes it should be removed.

File: README.md
Comment (rphair) [inline README.md:151]:
```suggestion
<p align="right"><i>Last updated on 2024-05-19</i></p>
```

**Review by rphair — COMMENTED** [review]
@kozross re: https://github.com/cardano-foundation/CIPs/pull/624#pullrequestreview-2039773732 thanks for https://github.com/cardano-foundation/CIPs/pull/624/commits/eaa2263e73037389fed474c406820c555e3cf337 but there's still a merge conflict due the modification time on that section also being changed:

**Comment by rphair** [comment]
@zliu41 apologies for extra tags above... I fat fingered some GitHub feature while my Internet connection was bad.  I understand you think this is ready to merge and FYI the editorial consensus from CIP meeting 2 days ago was that this is ready to go pending the administrative requirements that have recently been addressed.

**Review by Ryun1 — APPROVED** [review]

**Pull request merged**
